### ISSUE TITLE
Failing during GAF writing (src/MBG.cpp:99 and src/MBG.cpp:1435 depending on dataset)

### ISSUE THREAD

**Issue by biorover** [maickrau/MBG#15]
Hello, I'm trying to use MBG within the verkko pipeline, but it keeps failing while writing the GAF file (graph generation is fine and the graph is beautiful, but verkko obviously needs the GAF to continue). At first I was getting the following error:
```
MBG: src/MBG.cpp:1435: sortPaths(std::vector<ReadPath>&)::<lambda(const ReadPath&, const ReadPath&)>: Assertion `false' failed.
```
Which was maybe caused by identical (or nearly identical) reads? I finally got around this by splitting the reads into a bunch of subsets, running MBG on each separately, finding a bad chunk in the middle, and excluding it. I was then able to run MBG in single-K mode just fine. However, when I then tried to build a multiplex debruijn graph (adding -r 15000 -R 4000 as in verkko defaults), it failed with 
```
MBG: src/MBG.cpp:99: void updatePathRemaining(size_t&, size_t&, bool, const DumbSelect&, size_t): Assertion `expandedPoses.countOnes() >= overlap + 2' failed.
```
Any suggestions/help?

**Comment by biorover** [comment]
Update: a minor nudge to `-k` (from 1015 to 1003) and the cleaned dataset finished fine, so the second error must be some weird interaction between both the graph and the reads. Would still be nice to have both errors fixed so I don't have to manually run MBG and then massage the results back into verkko after finally making it work, though.

**Comment by maickrau** [comment]
Could you please share the bad reads chunk?

**Comment by biorover** [comment]
soon ;)

**Comment by maickrau** [comment]
There was a similar bug in a different dataset which was fixed in [ad2934f](https://github.com/maickrau/MBG/commit/ad2934f0dacaaf7ad21e3c0bd46da67f8a4e85ed). Could you please try again with the most recent commit

**Comment by biorover** [comment]
Sure enough, that fixed it. Thanks!!!
